framework/src/state_machine/api_context.ts

```diff
@@ -24,7 +24,7 @@ interface Params {
 export const createAPIContext = (params: Params) => new APIContext(params);
 
 export const createNewAPIContext = (db: StateDBReadWriter) =>
-	new APIContext({ stateStore: new PrefixedStateReadWriter(db), eventQueue: new EventQueue() });
+	new APIContext({ stateStore: new PrefixedStateReadWriter(db), eventQueue: new EventQueue(0) });
```

Comment:
why is it height `0` for this event queue?

Reply:
This is suggested by @ishantiw

Reply:
I would say take second param as height defaulting to 0 `createNewAPIContext = (db: StateDBReadWriter, height: number=0)` otherwise we can never instantiate `APIContext` with a correct height when needed